lib/frequency_table.rb

```diff
@@ -6,33 +6,48 @@
 require "overlap/ht_item_overlap"
 
 class FrequencyTable
-  def initialize
-    @table = Hash.new do |hash, member|
-      hash[member] = Hash.new { |fmt_hash, fmt| fmt_hash[fmt] = Hash.new(0) }
+  protected attr_reader :table
+
+  def initialize(data: nil)
+    @table = case data
+    when String
+      JSON.parse(data, symbolize_names: true)
+    when Hash
+      deep_copy data
+    when NilClass
+      {}
+    else
+      raise "unrecognized data format #{data.inspect}"
     end
   end
 
-  def organizations
-    @table.keys.sort
+  def to_json
+    JSON.generate(table)
   end
 
-  def [](organization)
-    @table[organization.to_sym]
+  def ==(other)
+    self.class == other.class && table == other.table
   end
 
-  # For testing. I don't know if this is useful in the long run.
-  def to_h
-    @table.clone.freeze
+  def fetch(organization: nil, format: nil, bucket: nil)
```

Comment:
This makes things much clearer in terms of understanding what the layers in the hash are